src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -11,73 +11,85 @@ import {
   PopoverAnchorPosition,
 } from '@elastic/eui';
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
-import { Query } from '../..';
+import { isEqual } from 'lodash';
 import { LanguageConfig } from '../../query';
 import { getQueryService } from '../../services';
 
 export interface QueryLanguageSelectorProps {
-  query: Query;
   onSelectLanguage: (newLanguage: string) => void;
   anchorPosition?: PopoverAnchorPosition;
   appName?: string;
 }
 
-const mapExternalLanguageToOptions = (language: LanguageConfig) => {
+interface LanguageOption {
+  label: string;
+  value: string;
+}
+
+const mapExternalLanguageToOptions = (language: LanguageConfig): LanguageOption => {
   return {
     label: language.title,
     value: language.id,
   };
 };
 
 export const QueryLanguageSelector = (props: QueryLanguageSelectorProps) => {
-  const [isPopoverOpen, setPopover] = useState(false);
-  const [currentLanguage, setCurrentLanguage] = useState(props.query.language);
-
   const queryString = getQueryService().queryString;
   const languageService = queryString.getLanguageService();
 
-  const datasetSupportedLanguages = useMemo(() => {
-    const dataset = props.query.dataset;
-    if (!dataset) {
-      return undefined;
-    }
-    const datasetType = queryString.getDatasetService().getType(dataset.type);
-    return datasetType?.supportedLanguages(dataset);
-  }, [props.query.dataset, queryString]);
+  const [isPopoverOpen, setPopover] = useState(false);
+  const [currentLanguage, setCurrentLanguage] = useState<string>(
```

Comment:
This works for now, but i dont like keeping track of another state here. This component should simply use the current state of the query service. Two issues can arise from this:
1. If the query state does not update, but the `setCurrentLanguage` does. They ar no out of sync again
2. If someone in fututre accidentally updates the query state based on the `currentLanguage`, we have a race condition again

Not a blocker, but we need to be figure out a nicer pattern for components that want to simply reflect the current state of the query and update it without causing update loops